pkg/chunk/chunk_test.go

```diff
@@ -133,9 +136,9 @@ const fixedTimestamp = model.Time(1557654321000)
 
 func TestChunkDecodeBackwardsCompatibility(t *testing.T) {
 	// Chunk encoded using code at commit b1777a50ab19
-	rawData := []byte("\x00\x00\x00\xb7\xff\x06\x00\x00sNaPpY\x01\xa5\x00\x00\x04\xc7a\xba{\"fingerprint\":18245339272195143978,\"userID\":\"userID\",\"from\":1557650721,\"through\":1557654321,\"metric\":{\"bar\":\"baz\",\"toms\":\"code\",\"__name__\":\"foo\"},\"encoding\":3}\n\x00\x00\x00\x15\x01\x00\x11\x00\x00\x01\xd0\xdd\xf5\xb6\xd5Z\x00\x00\x00\x00\x00\x00\x00\x00\x00")
+	rawData := []byte("\x00\x00\x00\xb7\xff\x06\x00\x00sNaPpY\x01\xa5\x00\x00\x04\xc7a\xba{\"fingerprint\":18245339272195143978,\"userID\":\"userID\",\"from\":1557650721,\"through\":1557654321,\"metric\":{\"bar\":\"baz\",\"toms\":\"code\",\"__name__\":\"foo\"},\"encoding\":3}\n\x00\x00\x00\x15\x01\x00\x11\x00\x00\x01У\xbe\xb3\xd5Z\x00\x00\x00\x00\x00\x00\x00\x00\x00")
 	decodeContext := NewDecodeContext()
-	have, err := ParseExternalKey(userID, "userID/fd3477666dacf92a:16aab37c8e8:16aab6eb768:38eb373c")
+	have, err := ParseExternalKey(userID, "userID/fd3477666dacf92a:16aab37c8e8:16aab6eb768:ff216d5b")
```

Comment:
Could you explain why this has changed? Given this is a backward compatibility test, we shouldn't touch it.

Reply:
> func dummyChunkForEncoding(now model.Time, metric labels.Labels, enc encoding.Encoding, samples int)
This is the function which is used for generating chunks for testing. It creates a chunk from `now-1 hour` to `now` but builds the samples starting from now, which means it would have samples with higher timestamp than chunks end time.

While most of the tests just add 1 sample, it works because it adds a sample for the last second of the chunks time range. I have fixed that behaviour which means it now adds samples for chunks from the start time of chunk, which also means it would create a different chunk.

I have built that rawData and signature from the same commit that is mentioned in the comments. Another option is to add a new function which would be just used in legacy tests building the same chunk and use the modified function everywhere else.

Reply:
I can second this. I ran into this issue previously when working on unit tests for a migration tool that was never merged.

Reply:
I have added code to build chunk to the same test function instead of sharing code with other tests to avoid such future encounters.